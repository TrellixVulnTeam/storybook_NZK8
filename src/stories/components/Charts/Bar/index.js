// @flow

import React, { PureComponent } from 'react';
import { BarChart, Bar, XAxis, YAxis, Tooltip, ResponsiveContainer, LabelList } from 'recharts';
import { ChartWrapper } from '../styled';
import { renderTooltip } from '../Tooltip';

type Props = {
  data: Array<string>,
};

class UpkBarChart extends PureComponent<Props> {
  /**
   * Render the Bar chart component
   * @return {React$Node}
   */
  render() {
    return (
      <ChartWrapper height="500px">
        <ResponsiveContainer>
          <BarChart barGap={20} data={this.props.data} margin={{ top: 40, right: 20, left: 20 }}>
            <XAxis
              dataKey="name"
              height={150}
              interval={0}
              fontSize={13}
              fontWeight="600"
              angle={-45}
              textAnchor="end"
              dy={15}
            />
            <YAxis type="number" domain={[0, 100]} />
            <Tooltip content={renderTooltip} />
            <Bar
              dataKey="value"
              fill="#007bff"
              radius={[6, 6, 0, 0]}
              label={{
                position: 'top',
                fontSize: 13,
                fontWeight: '600',
                formatter: value => {
                  return `${value > 0 ? value : '0'}%`;
                },
              }}
            >
              <LabelList
                dataKey="actualValue"
                position="bottom"
                fontSize={13}
                fill="#007bff"
                fontWeight="600"
              />
            </Bar>
          </BarChart>
        </ResponsiveContainer>
      </ChartWrapper>
    );
  }
}

export default UpkBarChart;
