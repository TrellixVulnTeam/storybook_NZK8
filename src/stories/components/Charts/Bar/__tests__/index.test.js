import React from 'react';
import { shallow } from 'enzyme';
import Bar from 'components/Charts/Bar/index';

describe('<Bar />', () => {
  it('should render correctly', () => {
    const sut = shallow(<Bar data={['0', '-1', '2', '3.5', '-4.5']} />);

    expect(sut).toMatchSnapshot();
  });
});
