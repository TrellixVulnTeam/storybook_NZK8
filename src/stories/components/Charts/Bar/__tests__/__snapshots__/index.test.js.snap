// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> should render correctly 1`] = `
<styled__ChartWrapper
  height="500px"
>
  <ResponsiveContainer
    debounce={0}
    height="100%"
    width="100%"
  >
    <BarChart
      barCategoryGap="10%"
      barGap={20}
      data={
        Array [
          "0",
          "-1",
          "2",
          "3.5",
          "-4.5",
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "left": 20,
          "right": 20,
          "top": 40,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
    >
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        angle={-45}
        dataKey="name"
        domain={
          Array [
            0,
            "auto",
          ]
        }
        dy={15}
        fontSize={13}
        fontWeight="600"
        height={150}
        hide={false}
        interval={0}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={false}
        scale="auto"
        textAnchor="end"
        tickCount={5}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        domain={
          Array [
            0,
            100,
          ]
        }
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        content={[Function]}
        contentStyle={Object {}}
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        separator=" : "
        useTranslate3d={false}
        viewBox={
          Object {
            "x1": 0,
            "x2": 0,
            "y1": 0,
            "y2": 0,
          }
        }
        wrapperStyle={Object {}}
      />
      <Bar
        animationBegin={0}
        animationDuration={400}
        animationEasing="ease"
        data={Array []}
        dataKey="value"
        fill="#007bff"
        hide={false}
        isAnimationActive={true}
        label={
          Object {
            "fontSize": 13,
            "fontWeight": "600",
            "formatter": [Function],
            "position": "top",
          }
        }
        layout="vertical"
        legendType="rect"
        minPointSize={0}
        onAnimationEnd={[Function]}
        onAnimationStart={[Function]}
        radius={
          Array [
            6,
            6,
            0,
            0,
          ]
        }
        xAxisId={0}
        yAxisId={0}
      >
        <LabelList
          dataKey="actualValue"
          fill="#007bff"
          fontSize={13}
          fontWeight="600"
          position="bottom"
          valueAccessor={[Function]}
        />
      </Bar>
    </BarChart>
  </ResponsiveContainer>
</styled__ChartWrapper>
`;
